woocommerce/woocommerce
# Fix HTTP 401 `woocommerce_rest_cannot_view` `Sorry, you cannot list resources` REST API response

### All Submissions:


### Changes proposed in this Pull Request:

Fixes an edge-case that causes a `woocommerce_rest_cannot_view` `Sorry, you cannot list resources` HTTP 401 error response to occur during REST API requests on sites where a plugin or code determines the current WordPress user before WooCommerce is fully initialized.

Based on suggestions and feedback from @ObliviousHarmony (thanks Christopher!)

Closes #26847.

### How to test the changes in this Pull Request:

Please see #26847 for suggested steps to reproduce the bug and test that this change fixes the issue.

### Other information:




### Changelog entry

> Fix - Unexpected HTTP 401 `woocommerce_rest_cannot_view` `Sorry, you cannot list resources` REST API responses that occur when a plugin or custom code determines the current WordPress user before WooCommerce is fully initialized.
